l1-contracts/contracts/bridge/ntv/INativeTokenVault.sol

```diff
@@ -17,20 +17,23 @@ interface INativeTokenVault {
     /// @notice The AssetRouter contract
     function ASSET_ROUTER() external view returns (IAssetRouterBase);
 
-    /// @notice Returns the total number of specific tokens locked for some chain
-    function chainBalance(uint256 _chainId, bytes32 _assetId) external view returns (uint256);
-
     /// @notice Returns if the bridged version of bridged token has been deployed
-    function isTokenNative(bytes32 assetId) external view returns (bool);
+    // function isTokenNative(bytes32 assetId) external view returns (bool);
+
+    /// @notice Returns the chain ID of the origin chain for a given asset ID
+    function originChainId(bytes32 assetId) external view returns (uint256);
 
-    /// @notice Used to register a token in the vault
+    /// @notice Registers tokens within the NTV.
+    /// @dev The goal is to allow bridging native tokens automatically, by registering them on the fly.
+    /// @notice Allows the bridge to register a token address for the vault.
+    /// @notice No access control is ok, since the bridging of tokens should be permissionless. This requires permissionless registration.
     function registerToken(address _l1Token) external;
 
     /// @notice Used to get the assetId of a token
     function getAssetId(uint256 _chainId, address _tokenAddress) external view returns (bytes32);
 
     /// @notice Used to get the the ERC20 data for a token
-    function getERC20Getters(address _token) external view returns (bytes memory);
+    function getERC20Getters(address _token, bool _legacy, uint256 _originChainId) external view returns (bytes memory);
```

Comment:
maybe I am missing something, but it seems like in all places where this function is used, _legacy = true

Reply:
Ok, then I did not do it properly, the idea is to have